src/main/java/org/icgc/argo/program_service/security/EgoSecurity.java

```diff
@@ -41,10 +40,9 @@ public EgoSecurity(@NonNull RSAPublicKey egoPublicKey) {
               .build(); //Reusable verifier instance
       val jwt = verifier.verify(jwtToken);
       return parseToken(jwt);
-    } catch (JWTVerificationException e) {
-      throw Status.PERMISSION_DENIED.asRuntimeException();
-    } catch (NullPointerException e) {
-      throw Status.UNAUTHENTICATED.asRuntimeException();
+    } catch (JWTVerificationException | NullPointerException e) {
+      // ego security shouldn't have knowledge of grpc
+      return Optional.empty();
```

Comment:
Hey @d8660091 , can you explain why you might not think it's a good idea to raise this exception?

If we end up agreeing, we should at least log with `log.error` so we don't silently suppress an exception.

Reply:
First JWTVerification means that a jwt is valid or no, it doesn't know if it's permitted or no. 

Second, when Optional is used, the function's error can be handled by return Optional.empty()

https://github.com/icgc-argo/program-service/pull/260/commits/c619761a8c0b6d58270840319bbf74529b05be47 shows changing this behavior makes the error more meaning ful

Reply:
Yea that is a good point about verification vs permission.